hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -145,7 +145,7 @@ protected HoodieWriteHandle(HoodieWriteConfig config, String instantTime, String
   private void initSecondaryIndexStats(boolean preserveMetadata) {
     // Secondary index should not be updated for clustering and compaction
     // Since for clustering and compaction preserveMetadata is true, we are checking for it before enabling secondary index update
-    if (config.isSecondaryIndexEnabled() && !preserveMetadata) {
```

Comment:
so the user can not disable the SI with write option now? what is the suggested way to disable it then?

Reply:
SI is expected to be created via spark-sql unlike other indexes we have. 
So, one has to go to spark sql to execute `drop index` for it to take effect. 

this is different from other indexes. 
for FILES, RLI, COL_STATS, writer config's value in each write could dictate whether to enable or disable the index of interest.